torch/sparse/__init__.py

```diff
@@ -132,3 +134,69 @@ def sum(input, dim=None, dtype=None):
             return torch._sparse_sum(input, dim, dtype=dtype)
         else:
             return torch._sparse_sum(input, dtype=dtype)
+
+
+def mul(input, other):
+    r"""
+    Each element of the SparseTensor :attr:`input` is multiplied by each element of the SparseTensor :attr:`other`.
+    The resulting SparseTensor is returned. The sparse_dim and dense_dim of :attr:`input` and :attr:`other`
+    must be the same. The each dense_dim size of :attr:`input` and :attr:`other` must be the same. The
+    sparse_dim shape of :attr:`input` and :attr:`other` must be broadcastable, e.g., given sparse_dim ``d``,
+    ``input.size(d) == other.size(d)`` or ``input.size(d) > other.size(d) and other.size(d) == 1``.
+    Autograd is supported, and gradients of :attr:`input` and :attr:`other` are coalesced.
```

Comment:
Question for my information: isn't this a true multiply? Because zero times zero is zero. (Whereas with div, I agree that it needs to be sparse.div specifically)

Reply:
I haven't got a chance to check the code, but I totally agreed with both of your points: 1) forward of sparse_mul is a true mul; 2) backward of masked locations do not have gradients, which is the only difference.

It is a bit different in div since forward of sparse_div should have no entry for div(a_nnz_loc, a_masked_loc).

I will take a careful look at the code and come back to this thread.